### PR TITLE
added file type check to doc extraction

### PR DESCRIPTION
**Pull request by Scott771**
Currently, if you use DocumentExtraction with a file that isn't a PDF (the only currently supported doc type), e.g. .docx, you get a nondescript 500 error when you check the jobstatus. Changed the code to instead ensure that they are only uploading permissible documents initially. 

Current error at JobStatus
```
{'code': 500,
 'error_type': 'CyclibException',
 'message': 'http://apiserver:5000/async/a376ed6f-a00f-47ea-862a-3b696f896169: {}',
 'content': {},
 'error_class': 'ServiceException'}
```

### CODE REVIEW

**Comment by jkgenser** [comment]
Can you add a test that triggers this?

File: indico/queries/documents.py [hunk 1]
```diff
@@ -80,6 +81,11 @@ def __init__(
         self.files = files
         self.json_config = json_config
         self.upload_batch_size = upload_batch_size
+        for f in self.files:
```
Comment (nkassis) [inline indico/queries/documents.py:84]: I'm concerned with this check, this assumes all the pdf will have .pdf at the end which might not be the case in all customer envs. It's a very crude way to check.

Comment (Scott771) [reply indico/queries/documents.py:84]: What do you think about something like below? The filetype.match(f) would potentially lead here to a FileNotFoundError- but I think that's probably better than also having that happen later as well. 
```
        for f in self.files:
            doctype = filetype.match(f)
            if doctype.MIME not in ('application/pdf',):
                raise IndicoError(
                    "Only PDF files can be used with DocumentExtraction. You provided: {f}"
                )
```

Comment (nkassis) [reply indico/queries/documents.py:84]: yeah that should work, as long as it's consistent on windows and unix hopefully

Comment (dzereski) [reply indico/queries/documents.py:84]: Shouldn't this work with TIF files as well?  Should also check ".tif", ".tiff"

**Review by jkgenser — COMMENTED** [review]
stale?

**Pull request closed without merging**